api/v1alpha1/labels.go

```diff
@@ -16,6 +16,12 @@ const (
 	ShardLabelKey             = "kargo.akuity.io/shard"
 	StageLabelKey             = "kargo.akuity.io/stage"
 
+	// MigratedLabelKey is a label set on a resource that has successfully been
+	// migrated, indicating that no further actions are required. This label is
+	// set to "true" when the migration is complete. An example of this is the
+	// migration of the ProjectSpec to a ProjectConfig resource.
+	MigratedLabelKey = "kargo.akuity.io/migrated"
```

Comment:
Would it be worth having the key be more specific? I'm thinking about the possibility of a resource that may undergo multiple different sorts of migrations over the course of many Kargo releases. If we use a key as generic as this, we'll have trouble identifying which migrations a resource is or is not in need of.

Reply:
How will we denote that a resource has undergone multiple different migrations over time?

If we wanted them comma separated or structured as JSON or something along those lines, we should probably introduce some helpers for setting/checking migration status.